src/Features/LanguageServer/Protocol/Features/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs

```diff
@@ -519,6 +511,96 @@ private static UnifiedSuggestedActionSet OrganizeRefactorings(Workspace workspac
                 title: null,
                 priority: GetUnifiedSuggestedActionSetPriority(actions.Max(a => a.CodeActionPriority)),
                 applicableToSpan: refactoring.CodeActions.FirstOrDefault().applicableToSpan);
+
+            // Local functions
+            async Task<IUnifiedSuggestedAction> GetUnifiedSuggestedActionSetAsync(CodeAction codeAction, TextSpan? applicableToSpan, TextSpan selection, CancellationToken cancellationToken)
+            {
+                if (codeAction.NestedCodeActions.Length > 0)
+                {
+                    _ = ArrayBuilder<IUnifiedSuggestedAction>.GetInstance(codeAction.NestedCodeActions.Length, out var nestedActions);
+                    foreach (var nestedAction in codeAction.NestedCodeActions)
+                    {
+                        var unifiedAction = await GetUnifiedSuggestedActionSetAsync(nestedAction, applicableToSpan, selection, cancellationToken).ConfigureAwait(false);
+                        nestedActions.Add(unifiedAction);
+                    }
+
+                    var set = new UnifiedSuggestedActionSet(
+                        categoryName: null,
+                        actions: nestedActions.ToImmutable(),
+                        title: null,
+                        priority: GetUnifiedSuggestedActionSetPriority(codeAction.Priority),
+                        applicableToSpan: applicableToSpan);
+
+                    return new UnifiedSuggestedActionWithNestedActions(
+                        workspace, codeAction, codeAction.Priority, refactoring.Provider, ImmutableArray.Create(set));
+                }
+                else
+                {
+                    var fixAllSuggestedActionSet = await GetUnifiedFixAllSuggestedActionSetAsync(codeAction,
+                        refactoring.CodeActions.Length, document, selection, refactoring.Provider,
+                        refactoring.FixAllProviderInfo, workspace, cancellationToken).ConfigureAwait(false);
+
+                    return new UnifiedCodeRefactoringSuggestedAction(
+                            workspace, codeAction, codeAction.Priority, refactoring.Provider, fixAllSuggestedActionSet);
+                }
+            }
+        }
+
+        // If the provided fix all context is non-null and the context's code action Id matches
+        // the given code action's Id, returns the set of fix all occurrences actions associated
+        // with the code action.
+        private static async Task<UnifiedSuggestedActionSet?> GetUnifiedFixAllSuggestedActionSetAsync(
```

Comment:
Code similar to the existing one for code fixes, but has enough differences that sharing it did not make sense: https://github.com/dotnet/roslyn/blob/c9d9e122cfff99ce9508a54897254d407b148f48/src/Features/LanguageServer/Protocol/Features/UnifiedSuggestions/UnifiedSuggestedActionsSource.cs#L194-L249